### ISSUE TITLE
Typing aliases are missing

### ISSUE THREAD

**Issue by badge** [branchenergy/lambda-handler#4]
Type aliases are currently missing in places, leading to crazy return signatures such as:

```python
Union[
    Callable[
        [Callable[Concatenate[Dict[str, Any], _P], Dict[str, Any]]],
        AwsEventRawCallable,
    ],
    Callable[
        [Callable[Concatenate[AwsEventType, _P], LambdaResponse]],
        AwsEventRawCallable,
    ],
]
```

**Comment by badge** [comment]
This is caused by https://github.com/python/mypy/issues/11855, in particular neither `Concatenate` nor `ParamSpec` are supported by TypeAlias in mypy.

**Comment by badge** [comment]
Fixed in 2.0.0
